utility/session.go

```diff
@@ -0,0 +1,138 @@
+package utility
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/utility/types"
+)
+
+type Session interface {
+	NewSession(sessionHeight int64, blockHash string, geoZone GeoZone, relayChain RelayChain, application modules.Actor) (Session, types.Error)
+	GetServiceNodes() []modules.Actor // the ServiceNodes providing Web3 to the application
+	GetFishermen() []modules.Actor    // the Fishermen monitoring the serviceNodes
+	GetApplication() modules.Actor    // the Application consuming the web3 access
+	GetRelayChain() RelayChain        // the chain identifier of the web3
+	GetGeoZone() GeoZone              // the geolocation zone where all are registered
+	GetSessionHeight() int64          // the block height when the session started
+}
+
+type RelayChain Identifier
+type GeoZone Identifier
+
+type Identifier interface {
+	Name() string
+	ID() string
+	Bytes() []byte
+}
+
+var _ Session = &session{}
+
+type session struct {
+	serviceNodes  []modules.Actor
+	fishermen     []modules.Actor
+	application   modules.Actor
+	relayChain    RelayChain
+	geoZone       GeoZone
+	blockHash     string
+	key           []byte
+	sessionHeight int64
+}
+
+func (s *session) NewSession(sessionHeight int64, blockHash string, geoZone GeoZone, relayChain RelayChain, application modules.Actor) (session Session, err types.Error) {
+	s.sessionHeight = sessionHeight
+	s.blockHash = blockHash
+	s.geoZone = geoZone
+	s.relayChain = relayChain
+	s.application = application
+	s.key, err = s.sessionKey()
+	if err != nil {
+		return
+	}
+	s.serviceNodes = s.findClosestXServiceNodes()
+	s.fishermen = s.findClosestYFishermen()
+	return s, nil
+}
+
+// use the seed information to determine a SHA3Hash that is used to find the closest N actors based
+// by comparing the sessionKey with the actors' public key
+func (s *session) sessionKey() ([]byte, types.Error) {
+	sessionHeightBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(sessionHeightBytes, uint64(s.sessionHeight))
+	blockHashBz, err := hex.DecodeString(s.blockHash)
+	if err != nil {
+		return nil, types.ErrHexDecodeFromString(err)
+	}
+	appPubKey, err := crypto.NewPublicKey(s.application.GetPublicKey())
+	if err != nil {
+		return nil, types.ErrNewPublicKeyFromBytes(err)
+	}
+	return s.concat(sessionHeightBytes, blockHashBz, s.geoZone.Bytes(), s.relayChain.Bytes(), appPubKey.Bytes()), nil
+}
+
+// uses the current 'world state' to determine the service nodes in the session
+// 1) get an ordered list of the public keys of service nodes who are:
+//    - actively staked
+//    - staked within geo-zone
+//    - staked for relay-chain
+// 2) calls `pseudoRandomSelection(serviceNodes, numberOfNodesPerSession)`
+func (s *session) findClosestXServiceNodes() []modules.Actor {
```

Comment:
In addition to the comment above around the structure of this function, I also think that rather than having `X` and `Y` in these function names, we should just pass in `N` and return a list of that size.

Reply:
So it seems like you don't like my style of using `human readable signature names` over proper code style for these issues

Reply:
Wondering if this is a personal preference. @deblasis WYT?

Reply:
Very similar feedback here too https://github.com/pokt-network/pocket/pull/303#discussion_r998769579

Reply:
https://github.com/pokt-network/pocket/pull/303#issuecomment-1284187289

@deblasis WYT?

Reply:
Offline we agreed that an explicit comment of demonstrable not implementable intention is the path forward